### PR TITLE
Add markdown admonition extension

### PR DESCRIPTION
**Pull request by Bituvo**
## **This PR is a prerequisite to #14287!**

This adds an extension into the MkDocs workflow which renders Github admonition boxes correctly.

> [!NOTE]
> This is an admonition box.

> [!WARNING]
> This is another admonition box!

> \[!NOTE] This is what I would look like without this PR.

### CODE REVIEW

**Review by wsor4035 — APPROVED** [review]
as per https://github.com/minetest/minetest/pull/14287#issuecomment-1902207841 ill review this, however as a triager, ill leave it up to a core dev ( @appgurueu ) if they want to count this as an approval or not (depends how the linked comment is interpreted). 

to get a few possible objections out of the way

> it isnt regular markdown, normal people might not see it rendered with a "standard"(this markdown after all) render. 

we already have some markdown extension added, additionally, this is designed to semi gracefully fall back if the extension isnt there

> this markdown extension is new, and may not gain marketshare/be abandoned/removed

well, thats always a possibility with anything, but its github, which has one of the more popular flavors of markdown already (yes, this is an extension, not a part of gfm), so that covers the marketshare. as for if it gets abandoned, it has a semi graceful fallback and is easy enough to search and replace with `[!NOTE]` with `NOTE:` for example

with possible objections out of the way, i think this is a decent extension to add, as it allows making things stand out for various reasons, and is similar to functionality asciidoctor already has

**Comment by Zughy** [comment]
Out of curiosity: won't such extension make us more dependent on GH, closing MT into GH's walled garden (and complicate a potential migration even more)?

**Comment by wsor4035** [comment]
> Out of curiosity: won't such extension make us more dependent on GH, closing MT into GH's walled garden (and complicate a potential migration even more)?

....no

**Comment by rollerozxa** [comment]
> Out of curiosity: won't such extension make us more dependent on GH, closing MT into GH's walled garden (and complicate a potential migration even more)?

The Python Markdown extension library added in this PR ([markdown-gfm-admonition](https://github.com/dahlia/markdown-gfm-admonition)) is an independent Python library that has no relation to GitHub apart from the fact that the Markdown format extension was originally introduced in GFM. If you have ever added a table or a fenced codeblock (with or without syntax highlighting) to a Markdown document then those are also GFM extensions which has been widely adopted by Markdown libraries.

Anyways this is an example of how they will look when rendered by MkDocs, I think it looks pretty good:

image

**Review by appgurueu — APPROVED** [review]
Fine by me. This PR surely doesn't break anything.

As has been mentioned already, the only downside to using admonitions is that this isn't "standard Markdown". However:

* They are very nice (for the reader). I used admonitions a lot in AsciiDoc, I miss them sorely in Markdown.
* [pandoc, the (IMO) number one markdown library, has support for these admonitions](https://pandoc.org/MANUAL.html#extension-alerts)
* There are various other free libraries (like the Python library used here) that implement support for these admonitions.
* Should we decide against admonitions in the future, it is fairly easy to preprocess the Markdown to remove these admonitions, turning them back into regular text.
* We already use GFM (GitHub-flavored Markdown) features that are not in CommonMark like tables, so a Markdown viewer that "only" supports CommonMark will already have issues rendering our Markdown.
* Even the "fallback" of a quote with "[!NOTE]" or similar is not the end of the world: It conveys the same information while making clear that your viewer does not support some necessary syntax. It's ugly, but still readable.

**Pull request merged**
